### PR TITLE
Add some safety to overmap/dynamic interaction

### PR DESCRIPTION
**Pull request by Dennok**
## About The Pull Request

This must prevent THIS 

https://user-images.githubusercontent.com/7734424/153983496-21271cf4-aa6e-4971-b657-13ddbbb27f4c.mp4

## Why It's Good For The Game

Less ship erasing.

## Changelog
:cl:
fix: fixed some ship erasing
/:cl:

### CODE REVIEW

File: voidcrew/code/modules/overmap/dynamic_events.dm [hunk 1]
```diff
@@ -32,19 +32,34 @@
...
 	remove_mapzone()
 
 /obj/structure/overmap/dynamic/proc/remove_mapzone()
 	if(mapzone)
 		mapzone.clear_reservation()
 		QDEL_NULL(mapzone)
 
+/obj/structure/overmap/dynamic/proc/remove_docks()
+	if(reserve_dock)
+		qdel(reserve_dock, TRUE)
```
Comment (SuperNovaa41) [inline voidcrew/code/modules/overmap/dynamic_events.dm:45]: do you need the `force = TRUE` here? if not you can just do `QDEL_NULL(reserve_dock)`

Comment (Dennok) [reply voidcrew/code/modules/overmap/dynamic_events.dm:45]: Dock can be deleted only by force deleting. Safety from unnecesary dock deleting by any means.

File: voidcrew/code/modules/overmap/dynamic_events.dm [hunk 2]
```diff
@@ -32,19 +32,34 @@
...
+		reserve_dock = null
+	if(reserve_dock_secondary)
+		qdel(reserve_dock_secondary, TRUE)
+		reserve_dock_secondary = null
+
 /obj/structure/overmap/dynamic/ship_act(mob/user, obj/structure/overmap/ship/simulated/acting)
+	if(concerned)
+		to_chat(user, "<span class='notice'>Here too high traffic, try some later.</span>")
```
Comment (SuperNovaa41) [inline voidcrew/code/modules/overmap/dynamic_events.dm]:
```suggestion
		to_chat(user, "<span class='notice'>Too much traffic, try again later!</span>")
```

File: voidcrew/code/modules/overmap/dynamic_events.dm [hunk 3]
```diff
@@ -209,25 +226,24 @@
...
 
-	if(reserve_dock)
-		qdel(reserve_dock, TRUE)
-		reserve_dock = null
-	if(reserve_dock_secondary)
-		qdel(reserve_dock_secondary, TRUE)
-		reserve_dock_secondary = null
+	concerned = TRUE //Prevent someone to act with this while it reloads
+
+	remove_docks()
```
Comment (SuperNovaa41) [inline voidcrew/code/modules/overmap/dynamic_events.dm:237]: Before your change this is supposed to be called regardless of if `mapzone` is null or not. Is there a reason you're changing it around?

**Review by SuperNovaa41 — CHANGES_REQUESTED** [review]

File: voidcrew/code/modules/overmap/dynamic_events.dm (hunk 3 above)
Comment (Dennok) [reply voidcrew/code/modules/overmap/dynamic_events.dm:237]: If it has no mapzone it shouldn't have docks.

Comment (Dennok) [reply voidcrew/code/modules/overmap/dynamic_events.dm:237]: If it has dock with no mapzone, something going wrong, but anyway on new level generation dock will be overwritten, and landing check for mapzone, and if there no mapzone it generate new mapzone with new docks.

File: voidcrew/code/modules/overmap/dynamic_events.dm (hunk 2 above)
Comment (Dennok) [reply voidcrew/code/modules/overmap/dynamic_events.dm]: Thanks, i dont wery good in this.

File: voidcrew/code/modules/overmap/dynamic_events.dm (hunk 3 above)
Comment (SuperNovaa41) [reply voidcrew/code/modules/overmap/dynamic_events.dm:237]: if someone docks in space, does that call mapload? this might be deleting docks from space

Comment (Dennok) [reply voidcrew/code/modules/overmap/dynamic_events.dm:237]: Call. I check. And it replace all turfs by new space turfs.

Comment (Dennok) [reply voidcrew/code/modules/overmap/dynamic_events.dm:237]: And for dock in space it create new obj

**Comment by SuperNovaa41** [comment]
Ok great, I'll test merge this in a bit and see how it goes

**Comment by SuperNovaa41** [comment]
Seemed fine

**Pull request merged**